src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -560,7 +562,12 @@ uct_cuda_ipc_query_devices(
         uct_md_h md, uct_tl_device_resource_t **tl_devices_p,
         unsigned *num_tl_devices_p)
 {
-    return uct_cuda_base_query_devices_common(md, UCT_DEVICE_TYPE_SHM,
+#if HAVE_CUDA_FABRIC
+    uct_device_type_t dev_type = UCT_DEVICE_TYPE_NET;
+#else
+    uct_device_type_t dev_type = UCT_DEVICE_TYPE_SHM;
+#endif
```

Comment:
what if we always define it as `NET`?

Reply:
@brminich We could do that but we should probably wait until rkey_unpack failures are gracefully handled as cuda_ipc would become eligible for all endpoints.